inc/zoo/swar/SWAR.h

```diff
@@ -62,18 +73,82 @@ struct SWAR {
         AllOnes = ~std::make_unsigned_t<T>{0} >> PaddingBitsCount, // Also constructed in RobinHood utils: possible bug?
         LeastSignificantBit = meta::BitmaskMaker<T, std::make_unsigned_t<T>{1}, NBits>::value,
         MostSignificantBit = LeastSignificantBit << (NBits - 1),
-        LeastSignificantLaneMask =
-            sizeof(T) * 8 == NBits ? // needed to avoid shifting all bits
-                ~T(0) :
-                ~(~T(0) << NBits),
+        LeastSignificantLaneMask = []() {
+            if constexpr (NBits < sizeof(T) * 8) {
+                return (T(1) << NBits) - 1;
+            } else {
+                return ~T(0);
+            }
+        }(),
         // Use LowerBits in favor of ~MostSignificantBit to not pollute
         // "don't care" bits when non-power-of-two bit lane sizes are supported
-        LowerBits = MostSignificantBit - LeastSignificantBit;
+        LowerBits = MostSignificantBit - LeastSignificantBit,
+        MaxUnsignedLaneValue = LeastSignificantLaneMask;
+
+    template <typename InputIt>
```

Comment:
Do you mean this is for any input iterator?
Do we have a unit test that uses input interators?
Also, unfortunately, `from_range` is not an appropriate identifier, since we are using iterator pairs, I believe that the convention emerged to name things that take a single-argument range as the suffix `range`

Reply:
Could you propose an alternative?

Reply:
After I wrote the comment I thought about it and thought that we can keep `from_range`, these functions that take a pair of iterators ought to have been named ranges from the start, then it is legitimate to retrofit the particle "range" into identifiers.